lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -206,6 +216,22 @@ protected function _getPortableTableColumnDefinition($tableColumn)
         return $column;
     }
 
+
+    /**
+     * Get a doctrine/mysql compatible column default from
+     * the new information_schema behaviour introduced in MariaDb >= 10.2.7.
+     *
+     * @param null|string $columnDefault default value as stored in information_schema for MariaDB >= 10.2.7
+     */
+    private function getMariaDb1027ColumnDefault(?string $columnDefault): ?string {
+        $defaultValue = ($columnDefault === 'NULL') ? null : $columnDefault;
+        if ($defaultValue !== null && ($defaultValue[0] ?? '') === "'") {
+            // Get an unquoted version
+            $defaultValue = preg_replace('/^\'(.*)\'$/', '$1', $columnDefault);
```

Comment:
If the value is surrounded by quotes, it's most probably has special characters escaped as well. Could you add a test for a default value of a string containing single quotes?

Reply:
I initially made it, but could not test it with from a doctrine table spec [see where the test should be made here](https://github.com/belgattitude/dbal/blob/248a05abaf72c5bc4488549a0bd143c124c66351/tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php#L329). Magic quoting happens somewhere. For me there was already an issue with quotes insides quotes. 

I can add the correct escaping based on the pure-sql tests [here](https://github.com/belgattitude/dbal/blob/248a05abaf72c5bc4488549a0bd143c124c66351/tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php#L377) (as I did before).

But as this P/R growed a lot (travis trusty, mariadb 10.2 travis, mariadb detection...), I feel more comfortable to do it in separate P/R once this one have been merged.  Same for the 'CURRENT_TIMESTAMP()'  vs 'CURRENT_TIMESTAMP' as I commented before. 

It will be much more clear and there will probably a lot to discuss. My point, this P/R should not break anything.